src/relayserver/HttpServer.ts

```diff
@@ -54,46 +46,52 @@ export class HttpServer {
 
   close (): void {
     console.log('Stopping relay worker...')
-    this.backend.stop()
+    this.relayService?.stop()
   }
 
-  // TODO: use this when changing to jsonrpc
-  async rootHandler (req: any, res: any): Promise<void> {
-    let status
-    try {
-      let res
-      // @ts-ignore
-      const func = this.backend[req.body.method]
-      if (func != null) {
-        res = await func.apply(this.backend, [req.body.params]) ?? { code: 200 }
-      } else {
-        // @ts-ignore
-        // eslint-disable-next-line @typescript-eslint/restrict-template-expressions
-        throw Error(`Implementation of method ${req.body.params} not found on backend!`)
-      }
-      status = jsonrpc.success(req.body.id, res)
-    } catch (e) {
-      let stack = e.stack.toString()
-      // remove anything after 'rootHandler'
-      stack = stack.replace(/(rootHandler.*)[\s\S]*/, '$1')
-      status = jsonrpc.error(req.body.id, new jsonrpc.JsonRpcError(stack, -125))
+  pingHandler (req: Request, res: Response): void {
+    if (this.relayService == null) {
+      throw new Error('RelayServer not initialized')
     }
-    res.send(status)
-  }
-
-  pingHandler (req: any, res: any): void {
-    const pingResponse = this.backend.pingHandler(req.query.paymaster)
+    const paymaster = req.query.paymaster
```

Comment:
any special reason to move the pingHandler's logic into the HttpServer code?

Reply:
You are misreading this change, it seems. `pingHandler` was always here, I made it accept type annotated parameters and I removed `rootHandler` as it seemed redundant. We wanted to switch to JSON-RPC but later decided it does not have any actual benefit when we have 3 methods.